lib/matplotlib/stackplot.py

```diff
@@ -107,9 +111,22 @@ def stackplot(axes, x, *args,
         first_line = center - 0.5 * total
         stack += first_line
 
+    if hide_empty:
+        if any(k in kwargs for k in ['edgecolor', 'interpolate', 'where']):
+            raise ValueError('hide_empty is not compatible with edgecolor, '
```

Comment:
It is probably worth putting in the code to sort out which of the conflicting keys it is.  The more information we can give the user in the error message the easier it will be for them to fix the problem, and (hopefully) the happier they will be :)